src/lib/y2storage/blk_device.rb

```diff
@@ -508,11 +508,7 @@ def filesystem_label
     # @see #udev_paths
     # @return [String, nil]
     def udev_full_label
-      label = filesystem_label
-
-      return nil if label.nil? || label.empty?
-
-      File.join("/dev", "disk", "by-label", label)
+      Filesystems::MountByType::LABEL.udev_name(filesystem_label)
```

Comment:
Now, methods like `#preferred_mount_by`, `#preferred_name` or `path_for_mount_by` do not take into account the device content. But methods `#udev_full_label` and `#udev_full_uuid` does. Maybe, those methods should not consider its content either. I.e., `#udev_full_label` should return nil by default (only luks could return something) and `#udev_full_uuid` should be nil too (only MD could return something). What do you think? Maybe this change does not pay off because it requires to adapt logic in some parts of the code.

Reply:
Yes, those old methods look inconsistent now. But changing them is simply too risky. 

To be honest, even the fact that `MountByType` contains the word "mount" in the name and that it lives under the `Filesystem` namespace looks wrong now.

Things to consider for the future.